### PR TITLE
(DOCSP-12990): Add missing parts of query engine page

### PR DESCRIPTION
**Pull request by cbush**
### Issue JIRA link:
https://jira.mongodb.org/browse/DOCSP-12990

### Docs staging link (requires sign-in on MongoDB Corp SSO):
https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/query-engine/ios/query-engine

### CODE REVIEW

File: source/ios/query-engine.txt [hunk 1]
```diff
@@ -94,15 +208,15 @@ values.
      - Evaluates to ``true`` if the left-hand numerical or date expression is greater than or equal to the right-hand numerical or date expression. For dates, this evaluates to ``true`` if the left-hand date is later than or the same as the right-hand date.
 
    * - ``in``
-     - Evaluates to ``true`` if the left-hand expression is in the right-hand list.
+     - Evaluates to ``true`` if the left-hand expression is in the right-hand list or string.
```
Comment (nathan-contino-mongo) [inline source/ios/query-engine.txt:211]: do you mean:
```suggestion
     - Evaluates to ``true`` if the left-hand expression is in the right-hand list of strings.
```
?

File: source/ios/query-engine.txt [hunk 2]
```diff
@@ -56,17 +155,32 @@ An **expression** consists of one of the following:
...
+             :language: swift
 
       .. tab::
-          :tabid: objective-c
+         :tabid: objective-c
+
+         .. literalinclude:: /examples/generated/code/start/QueryEngine.codeblock.models.m
+             :language: objective-c
+
+   You can set up the realm for these examples as follows:
```
Comment (nathan-contino-mongo) [inline source/ios/query-engine.txt:169]:
```suggestion
   You can set up the {+realm+} for these examples as follows:
```

Comment (nathan-contino-mongo) [reply source/ios/query-engine.txt:169]: "as follows" is a little vague. Could we change this to "with the following Realm object models" or something like that?

File: source/ios/query-engine.txt [hunk 3]
```diff
@@ -24,6 +24,111 @@ For an introduction to NSPredicate in general, See Apple’s
...
+one of the following:
+
+- The name of a property of the object currently being evaluated.
+- An operator and up to two argument expression(s).
+- A literal string, number, or date.
+
+Dot Notation
+~~~~~~~~~~~~
+
+When referring to an object property, you can use the **dot notation** to refer
```
Comment (nathan-contino-mongo) [inline source/ios/query-engine.txt:59]: I feel like the article isn't necessary here. What do you think?
```suggestion
When referring to an object property, you can use **dot notation** to refer
```

**Review by nathan-contino-mongo — APPROVED** [review]
LGTM aside from a couple of small wording suggestions.

**Comment by nathan-contino-mongo** [comment]
Also, you may want to do a little work to ensure that the iOS test suite continues to reliably run in CI :)

File: source/ios/query-engine.txt (hunk 1 above)
Comment (cbush) [reply source/ios/query-engine.txt:211]: No. I mean the property can be a list or a string (but not, say, a number). Open to suggestions.

Comment (nathan-contino-mongo) [reply source/ios/query-engine.txt:211]: No problem, seems like i misunderstood what was going on here.

**Review by nathan-contino-mongo — APPROVED** [review]
LGTM

**Comment by astigsen** [comment]
A few minor things missing:

In the list of comparison operators:
*  `=` is missing as alternative to `==`.
* `between` could really use an example. It may not be clear to users how to define a range. It should be written as `BETWEEN {a, b}`.

In the list of string comparison operators:
*  `=` is missing as alternative to `==`.
* missing `!=`

Set operators
* `SOME` is missing as alternative to `ANY`

**Pull request merged**